src/context/data.ts

```diff
@@ -7,5 +9,10 @@ export type DataContext<T> = (payload: T) => ResponseTransformer
  * Returns a GraphQL body payload.
  */
 export const data: DataContext<Record<string, any>> = (payload) => {
-  return json({ data: payload }, { merge: true })
+  return (res) => {
+    const prevBody = jsonParse(res.body) || {}
+    const nextBody = mergeRight(prevBody, { data: payload })
+
+    return json(nextBody)(res)
```

Comment:
We should add somewhere a message if the user has passed an object into res.body?

The `jsonParse` function will return undefined in this way, maybe we can add a message into the catch

Reply:
> if the user has passed an object into res.body?

This would violate the rule that the response body must always be set as a string. This affects both standard context utilities and the ones you may have created by yourself. With these changes `req.body` cannot be anything other than string or `null`.